### PR TITLE
Adds CodeQL query to check for insecure RequestValidationMode in ASP.NET

### PR DESCRIPTION
**Pull request by cldrn**
The `requestValidationMode` attribute in ASP.NET is used to configure built-in validations to protect applications against code injections. Downgrading or disabling this configuration is not recommended. The default value 4.5 is the only recommended value as previous versions only test a subset of requests.

This PR includes a CodeQL query to detect applications with insecure values for `requestValidationMode`:
https://docs.microsoft.com/en-us/dotnet/api/system.web.configuration.httpruntimesection.requestvalidationmode?view=netframework-4.8

### CODE REVIEW

**Comment by calumgrant** [comment]
Hi Paulino, Many thanks for your contribution, we really appreciate it. As you may have seen, I have added a few more commits to this PR in order to get it production ready for final review. I think this way would be easiest, in particular because external contributors don't have all the tools needed to author tests - we are working on that feature.

**Comment by jf205** [comment]
Hi @cldrn! I'll carry out a quick editorial review of the docs for this query later today.

File: csharp/ql/src/Security Features/CWE-016/ASPNetRequestValidationMode.qhelp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+
+	<overview>
+		<p>
+			The <code>requestValidationMode</code> attribute in ASP.NET is used to configure built in validation to
```
Comment (jf205) [inline csharp/ql/src/Security Features/CWE-016/ASPNetRequestValidationMode.qhelp]:
```suggestion
			The <code>requestValidationMode</code> attribute in ASP.NET is used to configure built-in validation to
```

File: csharp/ql/src/Security Features/CWE-016/ASPNetRequestValidationMode.qhelp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		<p>
+			Always set <code>requestValidationMode</code> to 4.5, or leave it at its default value.
+		</p>
+
+	</recommendation>
+	<example>
+
+		<p>
+			The following example shows the <code>requestValidationMode</code>
+			attribute set to the value 4.0 which disables some protections and
```
Comment (jf205) [inline csharp/ql/src/Security Features/CWE-016/ASPNetRequestValidationMode.qhelp]:
```suggestion
			attribute set to a value of 4.0, which disables some protections and
```

File: csharp/ql/src/Security Features/CWE-016/ASPNetRequestValidationMode.qhelp [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+
+	<overview>
+		<p>
+			The <code>requestValidationMode</code> attribute in ASP.NET is used to configure built in validation to
+			protect applications against code injections. Downgrading or disabling
+			this configuration is not recommended. The default value of 4.5
+			is the only recommended value as previous versions only test a subset of requests.
```
Comment (jf205) [inline csharp/ql/src/Security Features/CWE-016/ASPNetRequestValidationMode.qhelp]:
```suggestion
			is the only recommended value, as previous versions only test a subset of requests.
```

File: csharp/ql/src/Security Features/CWE-016/ASPNetRequestValidationMode.ql [hunk 4]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @name Insecure configuration for ASP.NET requestValidationMode
+ * @description Setting requestValidationMode to less than 4.5 disables built-in validations
```
Comment (jf205) [inline csharp/ql/src/Security Features/CWE-016/ASPNetRequestValidationMode.ql]:
```suggestion
 * @description Setting 'requestValidationMode' to less than 4.5 disables built-in validations
```
Code elements in the `@description` should be in single quotes.

**Comment by cldrn** [comment]
Thanks for the feedback and help!

**Pull request merged**